test/unit/reducers/project-state-reducer.test.js

```diff
@@ -110,25 +143,38 @@ test('onLoadedProject upload, with canSave true, prepares to save', () => {
     expect(resultState.loadingState).toBe(LoadingState.AUTO_UPDATING);
 });
 
-test('onLoadedProject with id shows with id', () => {
+test('onLoadedProject (LOADING_VM_FILE_UPLOAD, false, true) shows without id', () => {
     const initialState = {
-        loadingState: LoadingState.LOADING_VM_WITH_ID
+        loadingState: LoadingState.LOADING_VM_FILE_UPLOAD
     };
-    const action = onLoadedProject(initialState.loadingState, true, true);
+    const action = onLoadedProject(initialState.loadingState, false, true);
+    const resultState = projectStateReducer(initialState, action);
+    expect(resultState.loadingState).toBe(LoadingState.SHOWING_WITHOUT_ID);
+});
+
+test('onLoadedProject (LOADING_VM_FILE_UPLOAD, false, false), with project id, shows with id', () => {
```

Comment:
Not sure I understand what's going on in this test...

Reply:
the test names are not exactly very clear! Let me try revising them.

Reply:
Is the revised name more helpful?